src/background/tab-state.js

```diff
@@ -63,14 +76,15 @@ export default function TabState(initialState, onchange) {
 
   this.onchange = onchange || null;
 
-  /** Replaces the H state for all tabs with the state data
-   * from `newState`.
+  /**
+   * Replaces the H state for all tabs.
    *
-   * @param newState - A dictionary mapping tab ID to tab state objects.
+   * @param {TabStateMap} newState - A dictionary mapping tab ID to tab state objects.
    *                   The provided state will be merged with the default
    *                   state for a tab.
    */
   this.load = function (newState) {
+    /** @type {TabStateMap} */
     const newCurrentState = {};
     Object.keys(newState).forEach(function (tabId) {
       newCurrentState[tabId] = Object.assign(
```

Comment:
Could we replace `Object.assign` to the spread syntax `{...DEFAULT_STATE, ...newState[tabId]}`

Reply:
Yes. I'm going to leave that out of scope for this PR, but use of `Object.assign` is one of many pre-modern ES code patterns that can be updated whenever someone has the time.